### DISCUSSION TITLE
Cluster GCN implementation in pyg?

### DISCUSSION THREAD

**Discussion by Anarion-zuo** [pyg-team/pytorch_geometric#8231] [Q&A]
Hi! I'm new here.

I'm looking for implementations of the [Cluster GCN](https://arxiv.org/abs/1905.07953), and found [this](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.ClusterGCNConv.html#torch_geometric.nn.conv.ClusterGCNConv). In the paper, the model first cluster nodes, then forward node representation in a GNN layer. I don't see the clustering operation in [this](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.ClusterGCNConv.html#torch_geometric.nn.conv.ClusterGCNConv). Wonder where I can find it.

**Comment by EdisonLeeeee** [comment]
You can check out these examples:
+ https://github.com/pyg-team/pytorch_geometric/blob/master/examples/cluster_gcn_reddit.py
+ https://github.com/pyg-team/pytorch_geometric/blob/master/examples/cluster_gcn_ppi.py

The first step (graph partition/clustering) is implemented using `ClusterData`.
